control/tests/freqresp_test.py

```diff
@@ -508,3 +509,84 @@ def test_dcgain_consistency():
 
     sys_ss = ctrl.tf2ss(sys_tf)
     np.testing.assert_almost_equal(sys_ss.dcgain(), -1)
+
+
+# Testing of the singular_value_plot_function
+class TSys:
+    """Struct of test system"""
+    def __init__(self, sys=None, call_kwargs=None):
+        self.sys = sys
+        self.kwargs = call_kwargs if call_kwargs else {}
+
+    def __repr__(self):
+        """Show system when debugging"""
+        return self.sys.__repr__()
+
+
+@pytest.fixture
+def ss_mimo_t():
+    A = np.diag([-1/75.0, -1/75.0])
+    B = np.array([[87.8, -86.4],
+                  [108.2, -109.6]])/75.0
+    C = np.eye(2)
+    D = np.zeros((2, 2))
+    T = TSys(ss(A, B, C, D))
+    T.omega = 0.0
+    T.sigma = np.array([[197.20868123, 1.39141948]])
+    return T
+
+
+@pytest.fixture
+def ss_miso_t():
+    A = np.diag([-1 / 75.0])
+    B = np.array([[87.8, -86.4]]) / 75.0
+    C = np.array([[1]])
+    D = np.zeros((1, 2))
+    T = TSys(ss(A, B, C, D))
+    T.omega = 0.0
+    T.sigma = np.array([[123.1819792]])
+    return T
+
+
+@pytest.fixture
+def ss_simo_t():
+    A = np.diag([-1 / 75.0])
+    B = np.array([[1.0]]) / 75.0
+    C = np.array([[87.8], [108.2]])
+    D = np.zeros((2, 1))
+    T = TSys(ss(A, B, C, D))
+    T.omega = 0.0
+    T.sigma = np.array([[139.34159465]])
+    return T
+
+
+@pytest.fixture
+def ss_siso_t():
+    A = np.diag([-1 / 75.0])
+    B = np.array([[1.0]]) / 75.0
+    C = np.array([[87.8]])
+    D = np.zeros((1, 1))
+    T = TSys(ss(A, B, C, D))
+    T.omega = 0.0
+    T.sigma = np.array([[87.8]])
+    return T
+
+
+@pytest.fixture
+def tsystem(request, ss_mimo_t, ss_miso_t, ss_simo_t):
+
+    systems = {"ss_mimo": ss_mimo_t,
+               "ss_miso": ss_miso_t,
+               "ss_simo": ss_simo_t,
+               "ss_siso": ss_simo_t
+               }
+    return systems[request.param]
```

Comment:
I moved away from explicitly writing out every test system three times. See #588.

Reply:
Sorry I don't understand this. What do you mean?

Reply:
See below